crates/noirc_evaluator/src/ssa_refactor/opt/mem2reg.rs

```diff
@@ -122,29 +83,24 @@ impl PerBlockContext {
             }
         }
 
-        // Identify any arrays that are returned from this function
         if let TerminatorInstruction::Return { return_values } = block.unwrap_terminator() {
             for value in return_values {
-                if let Some(address) = self.try_const_address(*value, dfg) {
-                    self.alloc_ids_used_externally.insert(address.alloc_id());
-                }
+                assert!(!self.last_stores.contains_key(value), "Mutable vars are loaded before being returned - if this pattern changes, so do our safety assumptions.");
```

Comment:
I'd like to retain the ability to pass in or return mutable references to functions for implementing references later on. Does handling this cause any bugs that were present previously?